inst/rmarkdown/templates/template-name/skeleton/skeleton.Rmd

```diff
@@ -1,21 +1,41 @@
 ---
-title: "The post title"
-date: "YYYY-MM-DDTHH:MM:SS"
+params:
+  post_title: "The Post Title"
```

Comment:
why does the post_title need to be in two places? I think I am missing some info about how rmarkdown parameterized templates work.

Reply:
haha, good point.
What I want is to automatically generate `slug` based on `title` but I didn't know how to refer  to `title` on the `slug` element. That's why I chose `post_title` params, so that people only need to provide that one.
Please let me know if you have any other solutions.

Reply:
ah good question, I'll try to think about it :thinking:

Reply:
another idea is that we put the `slug` logic in `wp_post` function (if people provide `slug` in the YAML header manually, it will overwrite the function). In this way, we don't need `post_title` params ofc.

Reply:
I think it might make sense. in blogdown such logic lives in the addin. (sorry for the delay!!)

Reply:
I can try to do that. Should I put the `date` logic in the `wp_post()` as well?

Reply:
It seems quite complicated if we want to implement the same `slug` rules as WordPress...
https://core.trac.wordpress.org/browser/tags/5.5.1/src/wp-includes/formatting.php#L2223

Reply:
* reg date, as you prefer!
* reg the slug would https://github.com/hrbrmstr/slugify be a good approximation?

Reply:
Thanks @maelle 
https://github.com/hrbrmstr/slugify is not available for R > 4 tho, would it be a problem if we import this package?

Reply:
I think it's fine if we add the dependency too as there aren't many users of the package

Reply:
Thanks @maelle! 
I made the changes: both `date` and `slug` are moved to the `wp_post()` function.